### PR TITLE
CLDR-17298 Fix problem with unit exponents

### PR DESCRIPTION
**Pull request by macchiati**
This is a fix for the problem noted in https://unicode-org.atlassian.net/browse/CLDR-17298?focusedCommentId=177586

CLDR-17298

- [ ] This PR completes the ticket.

ALLOW_MANY_COMMITS=true

### CODE REVIEW

**Review by btangmu — APPROVED** [review]

**Comment by macchiati** [comment]
Hi, I did some wording cleanup; could someone re-review?

File: docs/ldml/tr35-general.md [hunk 1]
```diff
@@ -951,15 +951,17 @@ Some of the constraints reference data from the unitIdComponents in [Unit_Conver
     </ul></td></tr>
 
 <tr><td><a name='unit_constant' href='unit_constant'>unit_constant</a></td><td>:=</td>
-    <td>("1"[0-9]+ | [2-9][0-9]*)("e" ("1"[0-9]+ | [2-9][0-9]*))?
+    <td>[1-9][0-9]* ("e" [1-9][0-9]*)?
```
Comment (FrankYFTang) [inline docs/ldml/tr35-general.md:954]: this will allow "1" in "per-1" . Is that ok?

File: docs/ldml/tr35-general.md [hunk 2]
```diff
@@ -951,15 +951,17 @@ Some of the constraints reference data from the unitIdComponents in [Unit_Conver
...
             <ul><li>kilowatt-hour-per-100-kilometer</li>
                 <li>gallon-per-100-mile</li>
                 <li>per-200-pound</li>
                 <li>per-12</li>
             </ul></li>
-            <li><em>Note:</em> The number is an integer greater than one.</li>
+            <li><em>Constraint:</em> The numeric value of the unit constant must be an integer greater than one.</li>
+            <li><em>Note:</em> The normal interpretation of <code>e</code> is used, where 2e6 = 2×10⁶.</li>
             <li><em>Note:</em> The <code>e</code> notation is optional: per-100-kilometer and per-1e2-kilometer are equivalent unit_identifiers.</li>
+            <li><em>Note:</em> When constructing identifiers, exponents should be greater than 3 and multiples of 3, even though parsers must accept the wider range.</li>
```
Comment (FrankYFTang) [inline docs/ldml/tr35-general.md:964]: but in the previous line you have the example of "per-1e2-kilometer" 1e2 is not "exponents ... greater than 3 and multiples of 3", right?

File: docs/ldml/tr35-general.md (hunk 1 above)
Comment (macchiati) [reply docs/ldml/tr35-general.md:954]: That is disallowed by the constraint below.

File: docs/ldml/tr35-general.md (hunk 2 above)
Comment (macchiati) [reply docs/ldml/tr35-general.md:964]: Yes. They are equivalent. The following line makes it clear that the 100 is the preferred form in that case for _generators_ of unit identifiers, even thought 1e2 must be accepted by _parsers_ of unit identifiers.

**Review by pedberg-icu — APPROVED** [review]

**Review by FrankYFTang — APPROVED** [review]

**Review by richgillam — APPROVED** [review]
LGTM except for a couple comments...

**Comment by macchiati** [comment]
Merging now, but can clean up wording later

**Pull request merged**